api/turing/cluster/knative_service.go

```diff
@@ -91,14 +92,15 @@ func (cfg *KnativeService) buildSvcSpec(
 	labels map[string]string,
 ) *knservingv1.ServiceSpec {
 	// Set max timeout for responding to requests
-	timeout := int64(knativeSvcDefaults.RequestTimeoutSeconds)
+	timeout := DefaultRequestTimeoutSeconds
 
 	// Build annotations, set target concurrency of 1
 	annotations := map[string]string{
 		"autoscaling.knative.dev/minScale": strconv.Itoa(cfg.MinReplicas),
 		"autoscaling.knative.dev/maxScale": strconv.Itoa(cfg.MaxReplicas),
-		"autoscaling.knative.dev/target":   strconv.Itoa(cfg.TargetConcurrency),
-		"autoscaling.knative.dev/class":    knativeSvcDefaults.AutoscalingClass,
+		"autoscaling.knative.dev/metric":   cfg.AutoscalingMetric,
+		"autoscaling.knative.dev/target":   cfg.AutoscalingTarget,
```

Comment:
not related to the MR, CMMIW, the [default in knative ](https://knative.dev/docs/serving/autoscaling/concurrency/#soft-limit)for this is 100 but we are setting to 1 only, any reason or am I understanding something wrong?

Reply:
The default value of 1 that we use is inspired by kserve's default to create the underlying knative resources - it will allow each replica to only handle 1 request at a given time (which may be an overkill but it's a safe value).